src/web/src/app/space/add/component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable, Subject, of} from 'rxjs';
+import { startWith, map, debounceTime, distinctUntilChanged, takeUntil, switchMap } from 'rxjs/operators';
+import { MatOption } from '@angular/material/core';
+import { Router, NavigationExtras } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { SpaceService } from '../service';
+
+import { Space } from '../model';
+
+@Component({
+	selector: 'app-space-add',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css'],
+})
+
+export class SpaceAddComponent implements OnInit {
+	createForm: FormGroup;
+	controlAutocomplete = new FormControl('');
+
+	spaces: Space[] = [];
+
+	filteredSpaces: Observable<Space[]>;
+	selectedSpace: Space;
+	private readonly ngUnsubscribe = new Subject();
+	keyWasChanged: boolean = false;
+	creatingSpace: boolean = false;
+
+	constructor(
+		private router: Router,
+		private fb: FormBuilder,
+		private space: SpaceService
+	) {
+		this._createForm();
+	}
+
+	private translitFromRuToEn = {
+	"ё": "yo", "й": "i", "ц": "ts", "у": "u", "к": "k", "е": "e", "н": "n", "г": "g",
+	"ш": "sh", "щ": "sch", "з": "z", "х": "h", "ъ": "'", "ф": "f", "ы": "i", "в": "v", "а": "a", "п": "p", "р": "r", "о": "o", "л": "l", "д": "d", "ж": "zh", "э": "e", "я": "ya", "ч": "ch", "с": "s", "м": "m", "и": "i", "т": "t", "ь": "'", "б": "b",
+	"ю": "yu"
+	};
+
+	ngOnInit() {
+
+		this.filteredSpaces = this.controlAutocomplete.valueChanges.pipe(
+			startWith(''),
+			debounceTime(300), // Optional: debounce input changes to avoid excessive requests
+			distinctUntilChanged(), // Optional: ensure distinct values before making requests
+			switchMap(value => this.space.getSpaceListWith10ItemsByName(value || '').pipe(
+				map(apiResponse => apiResponse.results)
+			))
```

Comment:
Давай стараться использовать имена максимально короткие, но достаточные чтобы можно было понять для чего это.
```suggestion
			switchMap(value => this.space.getList(10, 0, value || '').pipe(
				map(res => res.results)
			))
```

Reply:
Готово.